tests/components/enphase_envoy/conftest.py

```diff
@@ -0,0 +1,110 @@
+"""Define test fixtures for Enphase Envoy."""
+import json
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+from homeassistant.components.enphase_envoy import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry, load_fixture
+
+
+@pytest.fixture(name="config_entry")
+def config_entry_fixture(hass: HomeAssistant, config, serial_number):
+    """Define a config entry fixture."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title=f"Envoy {serial_number}" if serial_number else "Envoy",
+        unique_id=serial_number,
+        data=config,
+    )
+    entry.add_to_hass(hass)
+    return entry
+
+
+@pytest.fixture(name="config")
+def config_fixture():
+    """Define a config entry data fixture."""
+    return {
+        CONF_HOST: "1.1.1.1",
+        CONF_NAME: "Envoy 1234",
+        CONF_USERNAME: "test-username",
+        CONF_PASSWORD: "test-password",
+    }
+
+
+@pytest.fixture(name="gateway_data", scope="session")
+def gateway_data_fixture():
+    """Define a fixture to return gateway data."""
+    return json.loads(load_fixture("data.json", "enphase_envoy"))
+
+
+@pytest.fixture(name="inverters_production_data", scope="session")
```

Comment:
Scope session seem bad. They dont need to remain in scope when all other tests for other components test. Rather wastefull for memory.

Reply:
https://github.com/home-assistant/core/pull/80499

Reply:
@elupus thanks for this comment, I never realized it has impact on memory